### PR TITLE
Stat/hist

### PR DESCRIPTION
**Pull request by jan-www**
generate histogram data for given input

### CODE REVIEW

File: src/stat/hist.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * `rightLim`: right limit to calculate, **default=max value of input data**
+ * @returns
+ * { steps; number[], counts: number[] }
+ * `steps`: array of giving the breakpoints between histogram cells,
+ * `counts`: array of number of data points in histogram cells
+ */
+export const getHistData = (xData: Tensor1D | number[], params?: {
+    bins?: number,
+    leftLim?: number,
+    rightLim?: number
```
Comment (yorkie) [inline src/stat/hist.ts:21]: How about a complete name `rightLimit`?

File: src/stat/hist.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }): {
+    steps: number[],
+    counts: number[],
+  } => {
+  const xArray = checkJSArray(xData, 'float32', 1) as number[];
+  const nData = xArray.length;
```
Comment (yorkie) [inline src/stat/hist.ts:27]: Why the name of an array length is called `*data`?

File: src/utils/validation.ts [hunk 3]
```diff
@@ -23,6 +23,14 @@ export function checkArray (array: Tensor | RecursiveArray<number>, dtype = 'any
   });
 }
 
+
+export function checkJSArray (array: Tensor | RecursiveArray<number>, dtype = 'any', ensureDimension = -1): number | number[] | number[][] | number[][][] | number[][][][] | number[][][][][] | number[][][][][][] {
```
Comment (yorkie) [inline src/utils/validation.ts:27]:
```suggestion
export function checkJSArray(array: Tensor | RecursiveArray<number>, dtype = 'any', ensureDimension = -1): number | number[] | number[][] | number[][][] | number[][][][] | number[][][][][] | number[][][][][][] {
```

File: src/utils/validation.ts [hunk 4]
```diff
@@ -23,6 +23,14 @@ export function checkArray (array: Tensor | RecursiveArray<number>, dtype = 'any
...
+  return tidy(() => {
+    const tensor = checkArray(array, dtype, ensureDimension);
```
Comment (yorkie) [inline src/utils/validation.ts:29]: Why does the `checkArray()` function return a tensor? It sounds like a validation function and it should return a boolean value IMO.

**Review by yorkie — CHANGES_REQUESTED** [review]
By the way, shall we use the complete name `Histogram` for this module?

File: src/stat/hist.ts (hunk 2 above)
Comment (jan-www) [reply src/stat/hist.ts:27]: number of data

File: src/stat/hist.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import { Tensor1D } from "@tensorflow/tfjs-core";
+import { checkJSArray } from "../utils/validation";
```
Comment (yorkie) [inline src/stat/hist.ts:2]:
```suggestion
import { checkJSArray } from '../utils/validation';
```

File: src/stat/hist.ts (hunk 2 above)
Comment (yorkie) [reply src/stat/hist.ts:27]: How about the complete `numberOfData`?

**Pull request merged**